sqlserver/datadog_checks/sqlserver/schemas.py

```diff
@@ -0,0 +1,417 @@
+
+    # Requests for infromation about tables are done for a certain amount of tables at the time
+    # This number of tables doesnt slow down performance by much (15% compared to 500 tables)
+    # but allows the queue to be stable.
+    TABLES_CHUNK_SIZE = 500
+    # Note: in async mode execution time also cannot exceed 2 checks.
+    MAX_EXECUTION_TIME = 10
+    MAX_COLUMNS_PER_EVENT = 100_000
+
+    def __init__(self, check, config):
+        self._check = check
+        self._log = check.log
+        self.schemas_per_db = {}
+        self._last_schemas_collect_time = None
+        collection_interval = config.schema_config.get('collection_interval', DEFAULT_SCHEMAS_COLLECTION_INTERVAL)
+        self._max_execution_time = min(
+            config.schema_config.get('max_execution_time', self.MAX_EXECUTION_TIME), collection_interval
+        )
+        e = is_affirmative(config.schema_config.get('enabled', True))
+        print(e)
+        super(Schemas, self).__init__(
+            check,
+            run_sync=True,
+            enabled=is_affirmative(config.schema_config.get('enabled', True)),
+            expected_db_exceptions=(),
+            # min collection interval is a desired collection interval for a check as a whole.
+            min_collection_interval=config.min_collection_interval,
+            dbms="sqlserver",
+            rate_limit=1 / float(collection_interval),
+            job_name="schemas",
+            shutdown_callback=self.shut_down,
+        )
+        base_event = {
+            "host": None,
+            "agent_version": datadog_agent.get_version(),
+            "dbms": "sqlserver",
+            "kind": "sqlserver_databases",
+            "collection_interval": collection_interval,
+            "dbms_version": None,
+            "tags": self._check.non_internal_tags,
+            "cloud_metadata": self._check._config.cloud_metadata,
+        }
+        self._data_submitter = SubmitData(self._check.database_monitoring_metadata, base_event, self._log)
+
+    def run_job(self):
+        self._collect_schemas_data()
+
+    def shut_down(self):
+        self._data_submitter.submit()
+
+    @tracked_method(agent_check_getter=agent_check_getter)
+    def _fetch_schema_data(self, cursor, db_name):
+        start_time = time.time()
+        schemas = self._query_schema_information(cursor)
+        for schema in schemas:
+            tables = self._get_tables(schema, cursor)
+            tables_chunks = list(get_list_chunks(tables, self.TABLES_CHUNK_SIZE))
+            for tables_chunk in tables_chunks:
+                schema_collection_elapsed_time = time.time() - start_time
+                if schema_collection_elapsed_time > self.MAX_EXECUTION_TIME:
+                    # TODO Report truncation to the backend
+                    self._log.warning(
+                        """Truncated data due to the effective execution time reaching {},
+                         stopped on db - {} on schema {}""".format(
```

Comment:
`Truncated data due to the effective execution time` -> `Truncated data due to the execution time`

Reply:
added your change, with 'reaching {}**s**.'